src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -787,6 +798,43 @@ mobile_apis::Result::eType CommandRequestImpl::PrepareResultCodeForResponse(
   return result_code;
 }
 
+mobile_apis::Result::eType CommandRequestImpl::PrepareResultCodeForResponse(
+    const ResponseInfo& first,
+    const ResponseInfo& second,
+    const ResponseInfo& third) {
+  SDL_LOG_AUTO_TRACE();
+  if (IsResultCodeUnsupported(first, second) ||
+      IsResultCodeUnsupported(second, first) ||
+      IsResultCodeUnsupported(third, second)) {
+    return mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  }
+  if (IsResultCodeWarning(first, second) ||
+      IsResultCodeWarning(second, first) ||
+      IsResultCodeWarning(third, second)) {
+    return mobile_apis::Result::WARNINGS;
+  }
+  // If response contains erroneous result code SDL need return erroneus
+  // result code.
+  hmi_apis::Common_Result::eType first_result =
+      hmi_apis::Common_Result::INVALID_ENUM;
+  hmi_apis::Common_Result::eType second_result =
+      hmi_apis::Common_Result::INVALID_ENUM;
+  hmi_apis::Common_Result::eType third_result =
+      hmi_apis::Common_Result::INVALID_ENUM;
+  if (!first.is_unsupported_resource) {
+    first_result = first.result_code;
+  }
+  if (!second.is_unsupported_resource) {
+    second_result = second.result_code;
+  }
+  if (!third.is_unsupported_resource) {
+    third_result = third.result_code;
+  }
+  mobile_apis::Result::eType result_code = MessageHelper::HMIToMobileResult(
+      std::max(std::max(first_result, second_result), third_result));
```

Comment:
@RomanReznichenkoLuxoft I think you can use `std::max` version which accepts initializer list like:
```suggestion
      std::max({first_result, second_result, third_result});
```